scripts/getChangelog.py

```diff
@@ -1,90 +1,570 @@
-
-def print_seen():
-    print("  [already seen] " + pr_details["title"] + " (#" + str(pr_details["number"]) + ")")
-
-print()
-prsSeen = set()
-filename = BASE + " - " + HEAD + " changelog.csv"
-outputFile = open(filename, 'w')
-outputFile.write("Type,Merge date,URL,Title,Author,Type,Functionality group\n")
-commits = requests.get("https://api.github.com/repos/" + REPO + "/compare/" + BASE + "..." + HEAD, headers={'Authorization': 'token ' + TOKEN}).json()
-numCommits = len(commits["commits"])
-for i in range(numCommits):
-    sha = commits["commits"][i]["sha"]
-    commit = commits["commits"][i]
-    print("Commit "+ str(i+1) + " of " + str(numCommits))
-    if "Merge pull request #" in commit["commit"]["message"] or "Merge branch" in commit["commit"]["message"]:
-        print("  [is merge commit]")
-        continue
-    pr_match = re.search(r'\(#(\d{4})\)', commit["commit"]["message"])
-    if pr_match:
-        pr_number = pr_match.group(1)
-        if pr_number in prsSeen:
-            print_seen()
-            continue
-        pr_details = requests.get("https://api.github.com/repos/" + REPO + "/pulls/" + pr_number, headers={'Authorization': 'token ' + TOKEN}).json()
-        outputFile.write("PR," + pr_details["merged_at"] + "," + pr_details["html_url"] + ",\"" + pr_details["title"] + "\"," + pr_details["user"]["login"] + "\n")
-        print("  " + pr_details["title"] + " (#" + str(pr_details["number"]) + ")")
-        prsSeen.add(pr_number)
-        continue
-    time.sleep(1.5) # Avoid rate limit
-    prs = requests.get("https://api.github.com/search/issues?q=repo:" + REPO + "+type:pr+is:merged+" + sha, headers={'Authorization': 'token ' + TOKEN}).json()
-    if len(prs["items"]) == 0:
-        outputFile.write("Commit," + commit["commit"]["committer"]["date"] + "," + commit["html_url"] + ",\"" + commit["commit"]["message"].splitlines()[0] + "\"," + commit["committer"]["login"] + "\n")
-        print("  [orphan] " + commit["commit"]["message"].splitlines()[0])
-        continue
-    pr_details = prs["items"][0]
-    if pr_details["number"] in prsSeen:
-        print_seen()
-        continue
-    outputFile.write("PR," + pr_details["pull_request"]["merged_at"] + "," + pr_details["html_url"] + ",\"" + pr_details["title"] + "\"," + pr_details["user"]["login"] + "\n")
-    print("  " + pr_details["title"] + " (#" + str(pr_details["number"]) + ")")
-    prsSeen.add(pr_details["number"])
-outputFile.close()
+# Define variables
+owner = os.getenv('OWNER', "AntennaPod") # The owner (organisation or user) of the repository
+repo = os.getenv('REPO', "AntennaPod") # The repository name
+token = os.getenv('GITHUB_API_TOKEN') # The GitHub API token [evnironment variable, otherwise user input]
+base_ref = os.getenv('BASE') # The base reference (release code or branch); point of reference [user input]
+head_ref = os.getenv('HEAD') # The head reference (release code or branch); environment containing the changes [user input]
```

Comment:
If this is always user input, I don't think it needs to be read from the environment variables